pkg/client/grafanacloud/client.go

```diff
@@ -0,0 +1,80 @@
+// Package grafanacloud provides an interface to the Grafana Cloud API.
+package grafanacloud
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+)
+
+const integrationsAPIURL = "https://integrations-api.grafana.net"
+
+// Client is a grafanacloud API client.
+type Client struct {
+	c      *http.Client
+	apiKey string
+}
+
+// NewClient creates a new Grafana Cloud client. All requests made will be
+// performed using the provided http.Client c. If c is nil, the default
+// http client will be used instead.
+//
+// apiKey will be used to authenticate against the API.
+func NewClient(c *http.Client, apiKey string) *Client {
+	if c == nil {
+		c = http.DefaultClient
+	}
+	return &Client{c: c, apiKey: apiKey}
+}
+
+// AgentConfig generates a Grafana Cloud Agent config from the given stack.
+// The config is returned as a string in YAML form.
+func (c *Client) AgentConfig(ctx context.Context, stackID string) (string, error) {
+	req, err := http.NewRequestWithContext(
+		ctx, "GET",
+		fmt.Sprintf("%s/stacks/%s/agent_config", integrationsAPIURL, stackID),
```

Comment:
What do you think about using `%s/stacks/%s/agent_config.yaml` instead? What logic should live in the client?

Reply:
Does that endpoint exist? I get a 404 when I try it. 

Given the Agent will be responsible for stitching together pieces of information in the future, I think it's fine for the logic to exist here. Plus it was faster for me to add client-side than to put it on your backlog with Evan.

Reply:
> Does that endpoint exist? I get a 404 when I try it.

No, we would have to add it. 

> Given the Agent will be responsible for stitching together pieces of information in the future,

I wonder how much logic the Agent must really have for the config. I'm afraid that if we go too far we block users from using their config management of choice. If the config consists of plain files users can manipulate them as they like. However, if they must use another tool it becomes more complicated. I'm probably missing a lot here. Do we have a document on future plans?

Reply:
Users will always be able to use whatever software they want to configure the Agent. This PR is just for the turn-key get-started install script on Grafana Cloud; we will never have a hard dependency on building a tool that must be used for configuring the Agent.